lib/widgets/home.dart

```diff
@@ -1,97 +1,539 @@
-                MessageListPage.buildRoute(context: context,
-                  narrow: const StarredMessagesNarrow())),
-              child: Text(zulipLocalizations.starredMessagesPageTitle)),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                InboxPage.buildRoute(context: context)),
-              child: const Text("Inbox")), // TODO(i18n)
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                SubscriptionListPage.buildRoute(context: context)),
-              child: const Text("Subscribed channels")),
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                RecentDmConversationsPage.buildRoute(context: context)),
-              child: Text(zulipLocalizations.recentDmConversationsPageTitle)),
-            if (testStreamId != null) ...[
-              const SizedBox(height: 16),
-              ElevatedButton(
-                onPressed: () => Navigator.push(context,
-                  MessageListPage.buildRoute(context: context,
-                    narrow: ChannelNarrow(testStreamId!))),
-                child: const Text("#test here")), // scaffolding hack, see above
-            ],
+      appBar: AppBar(),
+      body: const LoadingPlaceholder(),
+      floatingActionButton: TextButton(
+        onPressed: () => Navigator.push(context,
+          MaterialWidgetRoute(page: const ChooseAccountPage())),
+        child: Text(zulipLocalizations.chooseAccountPageTitle)));
+  }
+}
+
+class _NavigationBarButton extends StatelessWidget {
+  const _NavigationBarButton({
+    required this.icon,
+    required this.selected,
+    required this.onPressed,
+  });
+
+  final IconData icon;
+  final bool selected;
+  final void Function() onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    final iconColor = WidgetStateColor.fromMap({
+      WidgetState.pressed:  designVariables.iconSelected,
+      ~WidgetState.pressed: selected ? designVariables.iconSelected
+                                     : designVariables.icon,
+    });
+
+    return AnimatedScaleOnTap(
+      scaleEnd: 0.875,
+      duration: const Duration(milliseconds: 100),
+      child: IconButton(
+        icon: Icon(icon, size: 24),
+        onPressed: onPressed,
+        style: IconButton.styleFrom(
+          // TODO(#417): Disable splash effects for all buttons globally.
+          splashFactory: NoSplash.splashFactory,
+          highlightColor: designVariables.iconSelected.withValues(alpha: 0.05),
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(4))),
+        ).copyWith(foregroundColor: iconColor)));
+  }
+}
+
+void _showMenu(BuildContext context, {
```

Comment:
nit:
```suggestion
void _showMainMenu(BuildContext context, {
```

Makes it clearer up front which menu it is that this function shows.